background.js

```diff
@@ -70,31 +63,13 @@ function clean_utm(old_url) {
 
     }
 
-    console.debug("[clean_utm] returning " , ret_val['redirectUrl']);
+    console.debug("[link_cleaner] returning ", ret_val['redirectUrl']);
     return ret_val;
-}
+};
 
 function build_query_param_remover(shouldRemove) {
-    return function (requestDetails) {
-        var url = new URL(requestDetails.url);
-        if (url.search.length > 0) {
-            var params = url.searchParams;
-            var new_params = new URLSearchParams(params);
-            var needs_redirect = false;
-            for (let p of params.keys()) {
-                if (shouldRemove(p)) {
-                    needs_redirect = true;
-                    new_params.delete(p);
-                    //console.log("nuked query param", p);
-                }
-            }
-            if (needs_redirect) {
-                var new_url = new URL(url);
-                new_url.search = new_params.toString();
-                return { redirectUrl: new_url.href };
-            }
-        }
-        return {};
+    return function(requestDetails) {
+        return link_cleaner(requestDetails.url, shouldRemove);
```

Comment:
I've refactored your query params cleaner to be able to use it from `copy.js`, the js code that manages the right-click menu item